authentik/core/tasks.py

```diff
@@ -92,3 +95,56 @@ def clean_temporary_users(self: SystemTask):
             deleted_users += 1
     messages.append(f"Successfully deleted {deleted_users} users.")
     self.set_status(TaskStatus.SUCCESSFUL, *messages)
+
+
+@CELERY_APP.task(bind=True, base=SystemTask)
+@prefill_task
+def purge_password_history_table(self: SystemTask):
+    """Remove all entries from the core.models.UserPasswordHistory table"""
+    messages = []
+    try:
+        # n.b. a performance optimization to execute TRUNCATE
+        # instead of all().delete() would eliminate any FK checks.
```

Comment:
We could use a `TRUNCATE` command for performance, but I opted for `.all().delete()` to maintain conventions.

Reply:
Agreed. But it's in a task and we'll probably never have any issue with this regardless, so let's prioritize consistency.

Reply:
Yep, agreed. I'll remove this comment